csv/result.go

```diff
@@ -306,17 +323,36 @@ func (sfe *serializedFluxError) Error() string {
 // In case of an actual error:
 //   - if it's error that was serialized to CSV, it will be wrapped in serializedFluxError.
 //   - otherwise, it's a serialization error, it will be returned as-is.
-func readMetadata(r *csv.Reader, c ResultDecoderConfig, extraLine []string) (tableMetadata, error) {
+func readMetadata(r *bufferedCSVReader, c ResultDecoderConfig) (tableMetadata, error) {
 	n := -1
+	var recordStartIdx int
 	var resultID, tableID string
 	var datatypes, groups, defaults []string
-	for datatypes == nil || groups == nil || defaults == nil {
-		var line []string
-		if len(extraLine) > 0 {
-			line = extraLine
-			extraLine = nil
-		} else {
-			l, err := r.Read()
+	if c.NoAnnotations {
+		// No annotations means that we are going to treat all rows as part of the same result with exactly one table
+		resultID = "_result"
+		tableID = "0"
+		recordStartIdx = 0
+		line, err := r.Read()
```

Comment:
If the lines have differing numbers of fields, do we produce an error, or pad with nulls? I guess I will see as I read on.

Reply:
Thinking it over, since we still have a line for the column labels, that sets the number of columns, same as if we had annotations.